include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.hpp

```diff
@@ -0,0 +1,82 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_Trajectory_CoordinatesBroker__
+#define __OpenSpaceToolkit_Astrodynamics_Trajectory_CoordinatesBroker__
+
+#include <OpenSpaceToolkit/Core/Types/Integer.hpp>
+#include <OpenSpaceToolkit/Core/Types/String.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+
+using ostk::core::types::Integer;
+
+using ostk::astro::trajectory::CoordinatesSubset;
+
+/// @brief  State coordinates broker.
+
+class CoordinatesBroker
+{
+   public:
+    /// @brief              Constructor
+    ///
+    /// @code
+    ///                     CoordinatesBroker coordinatesSubset() ;
+    /// @endcode
+
+    CoordinatesBroker();
+
+    /// @brief              Adds a coordinates subset to be considered, returning the starting index it will occupy (or
+    /// that it occupies if it was already added) in the state coordinates
+    ///
+    /// @param              [in] aCoordinatesSubset a coordinate subset to be considered
+    ///
+    /// @return             The starting index of the subset in the state coordinates
+
+    Integer addSubset(const CoordinatesSubset& aCoordinatesSubset);
+
+    /// @brief              Checks if a coordinates subset has already been considered
+    ///
+    /// @param              [in] aCoordinatesSubset the coordinates subset to be checked
+    ///
+    /// @return             True if the coordinates subset is already considered
+
+    bool hasSubset(const CoordinatesSubset& aCoordinatesSubset) const;
+
+    /// @brief              Returns the starting index of a coordinates subset in the state coordinates
+    ///
+    /// @param              [in] aCoordinatesSubset the coordinates subset
+    ///
+    /// @return             The starting index of the subset int the state coordinates, throwing an exception if the
+    /// coordinates subset is not present
+
+    Integer getSubsetIndex(const CoordinatesSubset& aCoordinatesSubset) const;
+
+    /// @brief              Returns the total number of coordinates
+    ///
+    /// @return             The total number of coordinates
+
+    Integer getNumberOfCoordinates() const;
+
+    /// @brief              Returns the total number of coordinate subsets
+    ///
+    /// @return             The total number of coordinate subsets
+
+    Integer getNumberOfSubsets() const;
+
+   private:
+    Integer nextIndex_;
+    std::unordered_map<CoordinatesSubset, Integer> map_;
```

Comment:
There's a `Map` type in Core that you may be able to leverage here:
https://github.com/open-space-collective/open-space-toolkit-core/blob/main/include/OpenSpaceToolkit/Core/Containers/Map.hpp 

I think it would let you avoid the hashing stuff in CoordinatesBroker.hpp

Reply:
Oh God, thanks for this 👍

Reply:
I've tried, but I think it forces me to implemente the `<` operator between Subsets (which cannot be sorted as there is no notion of lesser/greater than between them).

Could it be that `Map.hpp` assumes an underlying sorted map?

Reply:
(I think it does include sorting):

<img width="227" alt="image" src="https://github.com/open-space-collective/open-space-toolkit-astrodynamics/assets/65550121/69f7d0e2-4b1b-4919-9bfe-41fbe14f056f">

Reply:
> Could it be that Map.hpp assumes an underlying sorted map?

Perhaps, but we also [`OrderedMap`](https://github.com/open-space-collective/open-space-toolkit-core/blob/main/include/OpenSpaceToolkit/Core/Containers/OrderedMap.hpp), so I think this one would naturally be unsorted.